vnext/JSI/Shared/ChakraRuntime.cpp

```diff
@@ -560,45 +546,68 @@ facebook::jsi::Function ChakraRuntime::createFunctionFromHostFunction(
   HostFunctionProxy *hostFuncProxy = new HostFunctionProxy(func, *this);
```

Comment:
>new [](start = 37, length = 3)

Maybe add a TODO here to use the same unique_ptr trick to avoid the manual new/delete calls